File: backend/app/models/ui_strings.py
from app.models.common import CamelModel


class UiStrings(CamelModel):
    """Fixed set of the app's own UI copy (buttons, placeholders, labels).
    English defaults here are the fallback shown before any document type
    (and therefore language) has been resolved."""

    subtitle: str = "Tell the assistant what agreement you need, then review and download it below."
    empty_state_placeholder: str = "Describe the agreement you need in the chat to get started."
    schema_load_error: str = "Couldn't load that document type. Please try again."
    chat_greeting: str = "Hi! What kind of legal document are you looking to put together today?"
    chat_thinking: str = "Assistant is thinking…"
    chat_error: str = "Something went wrong reaching the assistant. Please try again."
    retry: str = "Retry"
    send: str = "Send"
    message_label: str = "Message"
    download_pdf: str = "Download PDF"
    generating_pdf: str = "Generating PDF…"
    signing_note: str = "By signing below, each party agrees to enter into this agreement."
    signature_label: str = "Signature"
    print_name_label: str = "Print Name"
    title_label: str = "Title"
    company_label: str = "Company"
    notice_address_label: str = "Notice Address"
    date_label: str = "Date"
    term_fixed_label: str = "Expires"
    term_open_ended_label: str = "Continues until terminated"
    preview_footer: str = "Based on a Common Paper standard-terms template. Review with counsel before use."

File: frontend/lib/ui-strings.ts
export interface UiStrings {
  subtitle: string;
  emptyStatePlaceholder: string;
  schemaLoadError: string;
  chatGreeting: string;
  chatThinking: string;
  chatError: string;
  retry: string;
  send: string;
  messageLabel: string;
  downloadPdf: string;
  generatingPdf: string;
  signingNote: string;
  signatureLabel: string;
  printNameLabel: string;
  titleLabel: string;
  companyLabel: string;
  noticeAddressLabel: string;
  dateLabel: string;
  termFixedLabel: string;
  termOpenEndedLabel: string;
  previewFooter: string;
}

// English defaults — shown before any document type (and therefore
// language) has been resolved, and as the fallback for any key a
// translation call happens to omit.
export const DEFAULT_UI_STRINGS: UiStrings = {
  subtitle: "Tell the assistant what agreement you need, then review and download it below.",
  emptyStatePlaceholder: "Describe the agreement you need in the chat to get started.",
  schemaLoadError: "Couldn't load that document type. Please try again.",
  chatGreeting: "Hi! What kind of legal document are you looking to put together today?",
  chatThinking: "Assistant is thinking…",
  chatError: "Something went wrong reaching the assistant. Please try again.",
  retry: "Retry",
  send: "Send",
  messageLabel: "Message",
  downloadPdf: "Download PDF",
  generatingPdf: "Generating PDF…",
  signingNote: "By signing below, each party agrees to enter into this agreement.",
  signatureLabel: "Signature",
  printNameLabel: "Print Name",
  titleLabel: "Title",
  companyLabel: "Company",
  noticeAddressLabel: "Notice Address",
  dateLabel: "Date",
  termFixedLabel: "Expires",
  termOpenEndedLabel: "Continues until terminated",
  previewFooter: "Based on a Common Paper standard-terms template. Review with counsel before use.",
};
